docs/DeveloperGuide.md

```diff
@@ -155,6 +155,57 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+  * Returns an `AddEventCommand` to be executed by the `LogicManager`
+
+A successful execution of the AddEvent command is described as follows:
+1. The `AddEventCommand` retrieves the currently listed `Person`s from the `Model`.
+2. The `personToEdit` is obtained from the above list using the `Index` created during the parsing of the AddEvent command.
+3. `AddEventCommand` gets the `Schedule` of the `personToEdit` and creates a new `Schedule` with the added `Event` created during the parsing of the AddEvent command.
+4. `AddEventCommand` creates a new `Person` with the updated `Schedule` and uses `Model#setPerson()` to replace the `personToEdit` with the `Person` with the updated `Schedule`.
+5. `AddEventCommand` constructs the `CommandResult` and returns it to the `LogicManager`.
+
+  <img src="images/AddEventSequenceDiagram.png" width="1500" />
+
+#### Design considerations
+##### Differentiating recurring events and non-recurring events
+* **Alternative 1 (current choice):** Every `Event` has a `RecurFrequency` attribute. `RecurFrequency` has a `None` value to represent non-recurring tasks
+  * Pros: Easier to implement, no need to deal with Jackson library deciphering whether data should be recurring or non-recurring type
+* **Alternative 2:** Have a `RecurringEvent` class inherit from `Event` and only `RecurringEvent`s should have the `RecurFrequency` attribute
+  * Pros: Slightly more OOP
+  * Cons: 
+  1. Difficult to implement, increases likelihood of bugs and may take more time to implement should we misunderstand how Jackson library deciphers data.
+  2. More awkward type checking and casting when checking for recurrence.
+##### Implementing event's date, time and duration attributes
+* **Alternative 1 (current choice):** Use Java's in built LocalDate, LocalTime and Duration classes
+  * Pros: Easier to implement, no need to account for leap years, number of days in a month, formatting etc. Also has inbuilt support to calculate time.
+* **Alternative 2:** Create our own Date, Time and Duration classes
+  * Pros: More customisable
+  * Cons: Higher possibility of bugs if we do not properly account for leap years, number of days in a month, formatting etc. Will also take too much time to implement
+##### Updating the model when an event is added
+* **Alterative 1 (current choice):** Create a new `Schedule` and `Person` to update the `Model`
+  * Pros: More defensive
+  * Cons: Troublesome to implement
+* **Alternative 2:** Add an addEvent() function to `Schedule`, allows us to update the `Person` without creating objects
+  * Pros: Straightforward to implement
+  * Cons: In case a `Schedule` becomes unintentionally shared between 2 or more `Person`s, the editing of 1 `Person`'s `Schedule` when updating the `Model` may result in multiple `Person`s' `Schedule`s being edited at the same time, causing bugs to appear
```

Comment:
I think instead of numbering, we should continue with bulleting to keep consistency. And at this point, we might as well list all the pros and cons like this
```suggestion
* **Alternative 1 (current choice):** Every `Event` has a `RecurFrequency` attribute. `RecurFrequency` has a `None` value to represent non-recurring tasks
  * Pros: 
  Easier to implement, no need to deal with Jackson library deciphering whether data should be recurring or non-recurring type
* **Alternative 2:** Have a `RecurringEvent` class inherit from `Event` and only `RecurringEvent`s should have the `RecurFrequency` attribute
  * Pros: Slightly more OOP
  * Cons: 
  1. Difficult to implement, increases likelihood of bugs and may take more time to implement should we misunderstand how Jackson library deciphers data.
  2. More awkward type checking and casting when checking for recurrence.
##### Implementing event's date, time and duration attributes
* **Alternative 1 (current choice):** Use Java's in built LocalDate, LocalTime and Duration classes
  * Pros: Easier to implement, no need to account for leap years, number of days in a month, formatting etc. Also has inbuilt support to calculate time.
* **Alternative 2:** Create our own Date, Time and Duration classes
  * Pros: More customisable
  * Cons: Higher possibility of bugs if we do not properly account for leap years, number of days in a month, formatting etc. Will also take too much time to implement
##### Updating the model when an event is added
* **Alterative 1 (current choice):** Create a new `Schedule` and `Person` to update the `Model`
  * Pros: More defensive
  * Cons: Troublesome to implement
* **Alternative 2:** Add an addEvent() function to `Schedule`, allows us to update the `Person` without creating objects
  * Pros: Straightforward to implement
  * Cons: In case a `Schedule` becomes unintentionally shared between 2 or more `Person`s, the editing of 1 `Person`'s `Schedule` when updating the `Model` may result in multiple `Person`s' `Schedule`s being edited at the same time, causing bugs to appear
```